### PR TITLE
Add serviceAccountName input/search dropdown in deployment workload

### PR DESCRIPTION
**Pull request by jordojordo**
rancher/dashboard#1814 
Created form component to allow searching of service account names within the selected namespace. Currently, new service names must be created manually. If a user types in a service name that doesn't exist they will receive a message warning them.
Tested locally with multiple namespaces and service names.

### CODE REVIEW

File: components/form/ServiceNameSelect.vue [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+      const searched = e;
+
+      if (searched !== undefined && searched !== '') {
+        this.selected = searched;
+        this.$emit('input', searched);
+      }
+    },
+
+    clearSearch(event) {
+      if (event.pointerId === 1) {
```
Comment (nwmac) [inline components/form/ServiceNameSelect.vue:129]: I don't think pointerId is right here - I think we can remove this if

**Review by nwmac — CHANGES_REQUESTED** [review]
@jordojordo I updated your PR and changed the trash icon to a 'Clear' button. Also fixed an issue where the page would reload when you hit the clear button.

In addition to the comment, there is one other issue:

- The value fo serviceAccountName is not populated on edit. You can see this by editing a workload, setting a service account name and then editing the config of the workload - the service account name is empty instead of showing the correct name

**Comment by nwmac** [comment]
@jordojordo Can you update this PR?

File: components/form/ServiceNameSelect.vue [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+            :mode="mode"
+            :option-label="optionLabel"
+            :placement="$attrs.placement ? $attrs.placement : null"
+            :v-bind="$attrs"
+            @input="change"
+            @blur-create="blurCreate"
+          />
+        </div>
+      </div>
+      <button class="btn btn-sm clear-btn role-secondary col span-1" @click="clearSearch($event);">
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:149]: This shows up on the 'view config' page (so needs to take `mode` into account, and be hidden if is `_VIEW`)

File: components/form/LabeledSearchSelect.vue [hunk 3]
```diff
@@ -0,0 +1,378 @@
+<script>
```
Comment (richard-cox) [inline components/form/LabeledSearchSelect.vue:1]: There's a super duper amount of C&P from LabeledSelect in LabeledSearchSelect. The former has a `searchable` property, can that be used? If not can the functionality in the later be merged in to the former?

File: components/form/ServiceNameSelect.vue [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+        {{ t('generic.clear') }}
+      </button>
+    </div>
+    <template v-if="serviceNameNew">
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:153]: The initial view shows the `create it manually` warning. When there's no service supplied it shouldn't show
image

File: components/form/ServiceNameSelect.vue [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+  </div>
+</template>
+
+<style lang='scss' scoped>
+.clear-btn {
+  align-self: center;
+  height: 30px;
+}
+
+.input-container {
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:170]: This has all come from `InputWithSelect`, is it needed?

File: components/form/ServiceNameSelect.vue [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+      .vs__dropdown-toggle {
+        color: var(--primary) !important;
+        border-radius: var(--border-radius) 0 0 var(--border-radius);
+      }
+    }
+  }
+}
+
+.servicename-select ::v-deep {
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:246]: Is there a way to avoid using `::v-deep`?

File: components/form/ServiceNameSelect.vue [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+    focus() {
+      const comp = this.$refs.text;
+
+      if (comp) {
+        comp.focus();
+      }
+    },
+
+    change() {
+      const selectedOption = this.options.find(option => option.metadata.id === this.selected.serviceAccount);
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:86]: This throws an error whenever the user manually enters text

```
client.js?06a0:97 TypeError: Cannot read properties of null (reading 'serviceAccount')
    at eval (ServiceNameSelect.vue?0417:86)
```

File: components/form/ServiceNameSelect.vue [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+      <h3 class="col span-6">
+        {{ t('workload.serviceAccountName.label') }}
+      </h3>
+    </div>
+    <div class="row">
+      <div class="col span-5">
+        <div
+          :class="{ 'select-after': !selectBeforeText }"
+          class="servicename-select input-container container-flex"
+          @input="change"
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:125]: Is this needed when there's the `input` handler on `LabeledSearchSelect`?

File: components/form/ServiceNameSelect.vue [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+        >
+          <LabeledSearchSelect
+            v-if="selectLabel"
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:128]: This if is a bit odd, is it needed here?

File: components/form/ServiceNameSelect.vue [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+    <div class="spacer"></div>
+    <div class="row mb-10">
+      <h3 class="col span-6">
+        {{ t('workload.serviceAccountName.label') }}
+      </h3>
+    </div>
+    <div class="row">
+      <div class="col span-5">
+        <div
+          :class="{ 'select-after': !selectBeforeText }"
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:123]: Is this needed?

File: components/form/ServiceNameSelect.vue [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+
+  mixins: [labeledFormElement],
+
+  props: {
+    disabled: {
+      type:    Boolean,
+      default: false,
+    },
+
+    searchable: {
```
Comment (richard-cox) [inline components/form/ServiceNameSelect.vue:16]: The plumbing for this isn't passed through to `LabeledSearchSelect`

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: components/form/ServiceNameSelect.vue (hunk 4 above)
Comment (jordojordo) [reply components/form/ServiceNameSelect.vue:150]: That was a simple issue where `serviceNameNew` was searching for an empty string.

**Comment by jordojordo** [comment]
@richard-cox I have thoroughly cleaned this PR. Got rid of the LabeledSearchSelect component as it was completely unnecessary, I reused the LabeledSelect component to achieve the same results. Fixed the minor bugs that were all around, and removed the unused css.

**Review by richard-cox — CHANGES_REQUESTED** [review]
Two minor tweaks and one bug .. the 'Clear' feature doesn't get applied when editing a deployment with an existing value

**Review by richard-cox — APPROVED** [review]

**Pull request merged**